db/db_impl_compaction_flush.cc

```diff
@@ -1543,6 +1543,12 @@ Status DBImpl::WaitUntilFlushWouldNotStallWrites(ColumnFamilyData* cfd,
     WriteStallCondition write_stall_condition = WriteStallCondition::kNormal;
     do {
       if (write_stall_condition != WriteStallCondition::kNormal) {
+        // If the DB is in read-only mode and not coming back, no point waiting.
+        if (error_handler_.IsBGWorkStopped() &&
+            !error_handler_.IsRecoveryInProgress()) {
```

Comment:
It should be enough to check for IsBGWorkStopped(). There is no guarantee that a recovery, even if in progress, would finish in a timely manner. For example, if recovery is being attempted due to a NoSpace() error, it could go on forever if not enough space is freed.

Reply:
Good point, that should be consistent with how we stop user writes vs return errors to writers.